### ISSUE TITLE
[Bug] downloaded media files are not indexed to MediaStorage

### ISSUE THREAD

**Issue by djdisodo** [mozilla-mobile/fenix#8754]
downloaded media files are not indexed to [MediaStorage](https://developer.android.com/reference/android/provider/MediaStore)

## Steps to reproduce
download any image from web

### Expected behavior
shows up on gallery

### Actual behavior
nothing shows up

### Device information

* Android device: Android 9 (LOS 16)
* Fenix version: nightly(2020/02/25)
[](url)

**Comment by concastinator** [comment]
I can confirm this, I always have this problem with downloaded PDF files: when I open the Files application, instead of finding the file in "Downloads", I must look for it in "Internal Storage > Download". This doesn't make sense.

**Comment by MusicOnline** [comment]
I can confirm this too with photos and APKs (from what I've only tried). Basically you need to dig into Files app > Internal Storage > Download for anything and everything downloaded. It is certainly inconvenient.

**Comment by MrBrax** [comment]
Same here since i started using it in January. Very annoying.

**Comment by kglazko** [comment]
Could not reproduce the case of downloads only appearing in Internal Storage on a Nexus 5 or Pixel 3 @concastinator @MusicOnline . What devices were you seeing this happening on? On our devices, you can see the images/PDFs appearing in the Downloads section of Files, but not in the Images/Videos/Music areas of Files which this PR can certainly address (check the media type then also add to appropriate folder). We have a theory that the type of File Manager on some devices may be behaving differently..

QA, can you please check on various devices and see if you can find out which ones are reproducing this issue?

**Comment by catlee** [comment]
I'm on a Galaxy A5 and still see this issue. e.g. downloading PDFs or JPGs show up in the "Internal Storage -> Downloads" folder in the "My Files" app, but not in the "Downloads" folder.

**Comment by kglazko** [comment]
Thank you, I will try it out on a Samsung emulator.

**Comment by kglazko** [comment]
Seems like it could possibly be a Samsung issue, will wait for QA to check on different devices and see if they can narrow down a reproducible group of devices.

**Comment by concastinator** [comment]
@kglazko Mine is a Nokia 5 with up-to-date Android 9 (the Nokia 5 is part of the Android One program). The file manager is the default one, Google Files.

**Comment by MrBrax** [comment]
Happens on my Motorola One Zoom. There's no built in file manager so i use Amaze.

It shows up in Amaze but not in Discord/FB Messenger when files are downloaded.

**Comment by MusicOnline** [comment]
I'm using a Samsung Galaxy A20s. The pre-installed file manager has special Images, Videos, Audio, Documents, Downloads and APKs categories. I downloaded a `.pdf` which didn't appear in Documents, a `.jpg` which didn't appear in Images, a .`apk` which didn't appear in APKs and all of them, including an additional `.apkm` didn't appear in the Downloads category.

Additionally, I use Discord and the downloaded image was not registered to the in-app image quick upload menu, but is registered to the in-app file quick upload menu, if that gives any clue to the matter.

All of them are of course accessible through Internal Storage > Download. I tested using the latest Firefox Nightly available on the Google Play Store.

**Comment by rohshr** [comment]
This happens on my Samsung S7 as well. After downloading images, opening the image from the Download complete prompt opens the image alright and when I view the details, it shows the location as the Downloads folder. But most of the images don't always show up later in the Gallery or Google Photos.

The same issue with `.apk` files. Can run them directly from the browser after download but can't see them in the file manager.

It would be convenient to at least have a way to view the list of downloads in the browser to open the files.

**Comment by DrMcCoy** [comment]
Same on my Nokia 7.2 with (still) Android 9. I need to open up a file manager, go to the downloads folder and copy/move the file to a new location before it's accessible anywhere

**Comment by kglazko** [comment]
I also spent some time researching online and read on one site that the 'Downloads' folder in 'My Files' shows as history of downloads from approved apps such as Chrome, GDocs, etc. on Samsung. It may be a good idea to follow up with Samsung in the future to see we can get on such a whitelist if it exists @kbrosnan

**Comment by sflorean** [comment]
Investigated this on latest Nightly build and here are my findings:
- Nokia 6 (Android 7.1.1) - images don't show up in Gallery - downloads.
- Samsung Galaxy Note 10 (Android 10) - I was able to reproduce the behavior from [here](https://github.com/mozilla-mobile/fenix/issues/8754#issuecomment-614014433) only with the pdf files.
Also, on Samsung, the downloaded items are not displayed in the Downloads folder (pdf, video, audio, images) but are listed in Internal Storage - Downloads.

**Comment by MusicOnline** [comment]
Just updated through the Play Store and the MediaStorage indexing part seems fixed (seeing from images I have downloaded, not sure about other types). However, they're still not in the Downloads category in the Files app on my Samsung device. (Should this be covered in a different issue instead?)

**Comment by MrBrax** [comment]
> 
> 
> Just updated through the Play Store and the MediaStorage indexing part seems fixed (seeing from images I have downloaded, not sure about other types). However, they're still not in the Downloads category in the Files app on my Samsung device. (Should this be covered in a different issue instead?)

there hasn't been an update in a while on the play store, this still haven't been fixed. your device must have refreshed itself

**Comment by DrMcCoy** [comment]
Firefox Nightly (https://play.google.com/store/apps/details?id=org.mozilla.fennec_aurora) gets, as the name suggests, new builds pushed every night.

And yes, a downloaded image now turns up in other apps without me having to move or rename it in a file manager first. :)

**Comment by djdisodo** [comment]
still not working for me

**Comment by djdisodo** [comment]
@MusicOnline 
samsung a20 's android Q update was April 27, 2020
it could fixed it
[AbstractFetchDownloadService.kt#L465](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L465)
fenix uses mediastore but only for higher than android Q

**Comment by MusicOnline** [comment]
> @MusicOnline 
> samsung a20 's android Q update was April 27, 2020
> it could fixed it
> [AbstractFetchDownloadService.kt#L465](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L465)
> fenix uses mediastore but only for higher than android Q

True, I just got the A10 update a few days ago, didn't correlate it at the time.

**Comment by djdisodo** [comment]
[mozilla-mobile/android-components#6719](https://github.com/mozilla-mobile/android-components/pull/6719)

**Comment by Amejia481** [comment]
With the latest changes this should be fixed :)

**Comment by djdisodo** [comment]
seems fixed

**Comment by Amejia481** [comment]
@djdisodo Thanks for testing, we are waiting for the QA team to verify as they have a large set of devices.

**Comment by sflorean** [comment]
Tested this on Nightly 6/9 and based on [comment](https://github.com/mozilla-mobile/fenix/issues/8754#issuecomment-617812188) I can confirm that the issue is fixed for Nokia 6 (Android 7.1.1).

For Samsung Galaxy Note 10 (Android 10) - the downloaded items **are not displayed** in the Downloads folder (pdf, video, audio, images) but are listed in Internal Storage - Downloads.

On LG G7 FIT (Android 8) - all the files were displayed in "Downloads" folder. 

On Android 5&6 we couldn't check due to this [issue](https://github.com/mozilla-mobile/fenix/issues/11357).

@Amejia481 we will investigate on other devices and leave a comment.

**Comment by vesta0** [comment]
@Amejia481 are you still working on this?

**Comment by Amejia481** [comment]
No right now, as it has a lower priority, as after [this patch](https://github.com/mozilla-mobile/android-components/issues/7230), the issue is only reproducible in some specific devices

**Comment by Amejia481** [comment]
@vesta0 do we want to bump the priority on this issue?

**Comment by MrBrax** [comment]
It has been working for me for a few weeks now on a Motorola One Zoom

**Comment by kglazko** [comment]
@Amejia481 @sflorean Just to verify, it is still not working for Samsung devices? Or just the Galaxy Note 10? That was one of our main concerns because we have a significant amount of Samsung users and there was a question at one point of whether we would need to be whitelisted by them in order to show up in the Downloads section of their file manager.

**Comment by MusicOnline** [comment]
@kglazko Yep, downloads still aren't going to the Downloads category in the Samsung file manager app.

**Comment by MusicOnline** [comment]
Umm, there seems to be a regression? Media files downloaded are not registered to respective categories in Samsung's File Manager since some Nightly releases ago. Currntly, I'm on Nightly 200816 06:01.

I'm on Android 10 on a Samsung Galaxy A20s
device.

**Comment by evilpie** [comment]
Is there a list of affected devices? It seems to be a common complaint on reddit: (sample from the last few days)
- https://www.reddit.com/r/firefox/comments/inhm77/android_v801_downloaded_images_dont_imediately/
- https://www.reddit.com/r/firefox/comments/imkufg/wherere_my_downloads/
- https://www.reddit.com/r/firefox/comments/imnoww/where_are_my_downloads_android/
- https://www.reddit.com/r/firefox/comments/inzeho/image_downloads_in_firefox_android_arent_showing/
- https://www.reddit.com/r/firefox/comments/inuifd/issue_with_downloading_videos_with_firefox_for/

**Comment by yoasif** [comment]
OnePlus 6 user seeing this: https://www.reddit.com/r/firefox/comments/iomvmu/i_need_a_replacement_for_firefox_mobile/

**Comment by Amejia481** [comment]
> Is there a list of affected devices? It seems to be a common complaint on reddit: (sample from the last few days)
> 
>     * https://www.reddit.com/r/firefox/comments/inhm77/android_v801_downloaded_images_dont_imediately/
> 
>     * https://www.reddit.com/r/firefox/comments/imkufg/wherere_my_downloads/
> 
>     * https://www.reddit.com/r/firefox/comments/imnoww/where_are_my_downloads_android/
> 
>     * https://www.reddit.com/r/firefox/comments/inzeho/image_downloads_in_firefox_android_arent_showing/
> 
>     * https://www.reddit.com/r/firefox/comments/inuifd/issue_with_downloading_videos_with_firefox_for/

@evilpie that's sounds like a great idea, so far we don't have a list itself just the reports on this issue, having a list could be a great tool to identify the issue per manufacturer and OS version. Maybe our QA team can help us. Thanks for the suggestion.

**Comment by fersosam** [comment]
I have this problem on at least two devices. I just noticed it on the first device about two days ago. Chrome has no problems (the only other browser installed). If I reset the device the files that had already been downloaded but not index are finally indexed. But any new download after reboot is no longer indexed again, a new reboot is required to index the latest downloads. 

The first device has Android Firefox 80.1.2 (Build #2015761283) and is a Samsung Galaxy S20 Plus with One UI 2.5, Android 10, Kernel 4.19.87-19350896, and update G985FXXU4BTH5 / G985FOWA4BTH6 / G985FXXU4BTH5 (august 1, 2020). 

The second device has Android Firefox 80.1.2 (Build #2015761281) and is a Samsung Galaxy Tab A (2019)  with One UI 2.1, Android 10, Kernel 4.4.177-19000394, and update T510XXU3BTFN / T5100WO3BTG1 / (july 1, 2020).

Both devices are updated as much as Samsung and the Google Play store allow.

On the S20+ I downloaded Firefox Beta and had the same problem (I already uninsulated it, I don't remember exact version and build, sorry).

Hope this helps.

**Comment by abodea** [comment]
I re-tested this issue on the latest **Nightly build** for more details please check the following:

Note that the downloaded media files are inconsistent depending on the devices used.

| Device  | Android v | Images | Videos | Audio | Pdf   | 
| ------------- | :-------------: |  :-------------: |  :-------------: |  :-------------: |  :-------------: | 
| Google Pixel 4 XL              |          11        |    ✅               |         ❌      |       ❌         |      ✅           |     
|     Samsung Galaxy S10+           |     10             |       ✅            |     ❌       |     ❌           |      ✅            |     
|    Samsung Galaxy Tab S6     |        9          |         ✅          |      ❌       |     ❌           |           ✅       |       
|      HTC 10         |         8         |        ✅           |          ✅    |     ✅         |        ✅         |       
|      Samsung Galaxy S9      |         8       |        ✅           |         ✅     |     ✅          |        ✅         |    
|      Sony Xperia Z5       |         7       |        ✅           |         ✅     |     ✅          |        ✅         |       

❌ - the downloaded media file is not displayed in MediaStorage
✅ - the downloaded media file is displayed in MediaStorage

**`  Build : Nightly 200910 06:07 (Build #2015763057) AC: 58.0.20200908130811, 52ac53ba1 GV: 82.0a1-20200907094115 AS: 61.0.13 `**
I do believe this issue is regarding `One UI`.

@Amejia481 Please check the following GIF, note that I did 2 downloads with the latest Nightly and Brave browsers, you can see that for Nightly the `audio` file was not displayed.
![20200910-160901](https://user-images.githubusercontent.com/42831109/92738013-42319600-f384-11ea-9109-ab2126d9e985.gif)

Removing `eng:qa:needed` until further notice.

**Comment by Amejia481** [comment]
Thanks @abodea :)

**Comment by bmn-m** [comment]
Can confirm for pictures, videos and music on One Plus Nord Android 10.  Also this happens with FF Daylight.

**Comment by f00had** [comment]
I'm having the same issue in Firefox Nightly Build #2015763315 on my Galaxy Z Flip 5G (SM-F707B) running Android 10.

**Comment by yoasif** [comment]
[More devices -](https://www.reddit.com/r/firefox/comments/iqqha3/where_are_my_downloads_on_firefox_nightly_for/)

Note 9 pro xiaomi
Samsung Note10+
Note 20
Huawei P30 Pro

**Comment by anotherus3r** [comment]
Also doesn't work on Huawei devices.

**Comment by fersosam** [comment]
UPDATE: Ok, so a few days ago there was an update for firefox on my phone  (Google Play Store), sorry I hadn't had time to post this. It went from Build #2015761283 to Build #20157612835. The problem persists.

Also I've been intermittently trying with Nightly, I've tried with three iterations of Nightly. First with 200910 06:07 (Build #2015763059). Then 200911 14:03 (Build #2015763315). Just right now I tried with 200914 06:05 ( Build #2015763827). No joy with any of them. 

Again, I'm on a Samsung Galaxy S20+.

**Comment by Attoy** [comment]
Actually I've got the very same issue on a Samsung Galaxy A6+, Samsung Galaxy S20+, Samsung Galaxy Note 8, Samsung Galaxy S9 and Samsung Galaxy S8

**Comment by anotherus3r** [comment]
Should the severity maybe be higher since it affects users of a wide(r) range of smartphone models and OS versions of the two biggest smartphone vendors (Samsung and Huawei)?

Market share sources: [Statista](https://www.statista.com/statistics/271496/global-market-share-held-by-smartphone-vendors-since-4th-quarter-2009/), [IDC](https://www.idc.com/promo/smartphone-market-share/vendor), [Counterpoint](https://www.counterpointresearch.com/global-smartphone-share/)

**Comment by Amejia481** [comment]
> Should the severity maybe be higher since it affects users of a wide(r) range of smartphone models and OS versions of the two biggest smartphone vendors (Samsung and Huawei)?
> 
> Market share sources: [Statista](https://www.statista.com/statistics/271496/global-market-share-held-by-smartphone-vendors-since-4th-quarter-2009/), [IDC](https://www.idc.com/promo/smartphone-market-share/vendor), [Counterpoint](https://www.counterpointresearch.com/global-smartphone-share/)

Thanks @anotherus3r,  I agree, I changed to `S2`

**Comment by Djfe** [comment]
affected: OnePlus 3T, Android 9 (custom rom)

**Comment by GRArmstrong** [comment]
Also affected: Google Pixel 3a, Android 11. (Stable and Nightly)

**Comment by FoxTrotte** [comment]
I am having the same issue with images (I haven't checked other media types) on my Oneplus 6 running Oxygen Os 10.3.5, Android 10

**Comment by Croman12** [comment]
This issue also exists on my Xiaomi MI 9T with Android 10 (Havoc). If I download nzb files and try to upload them to my download program, the file is not there in my 'recent files' section. I manually have to go to the internal storage - - > Downloads-folder to select the nzb file. Definitely not only Samsung or Huawei devices are affected by this bug.

**Comment by drathier** [comment]
Firefox seems to download files to /Download/Download/filename, which is a folder I cannot find anywhere in the file tree using root browser. Chrome puts downloaded files in /storage/emulated/0/Download/filename, which works as expected. It did work as expected a couple of months back, and I'm on the verge of switching browsers now because of this.

**Comment by Djfe** [comment]
for me files are downloaded to the usual folder /storage/emulated/0/Download
(or /sdcard/Download but that's just a link)

Since files are downloaded correctly for everyone here, your issue seems to be a different one. Sorry to say this, but you are on a custom rom, right?
How do you know it's /Download/Download ?
Are you able to open files from Firefox directly (purple pop up)after finishing the download?

**Comment by drathier** [comment]
I'm on the standard samsung s10 rom, nothing custom going on. Should I open another ticket?

**Comment by payton-quinn** [comment]
+1 

Been having this problem as well.  It'd be nice if firefox also provided the option to view the downloaded file in Files on long press (like "View in Explorer" on windows).  Currently you can open it in photos from firefox and then use the share button to send it elsewhere, but that shouldn't be the only option.

**Comment by Kimarnic** [comment]
Still happening, samsung a50, what are you doing Mozilla? Stop adding features and fix it

**Comment by payton-quinn** [comment]
For additional context I'm using Firefox stable on a Google Pixel 3XL with Android 11.

**Comment by spoodbeast** [comment]
I just switched to a Galaxy S20 and am also having this issue, downloaded images and such through firefox only show up in astro file viewer and nothing else apparently, sometimes after a while they pop up in gallery but I don't know what triggers it. Just switching to the default samsung internet browser for now.

**Comment by Djfe** [comment]
you can manually trigger a scan by toggling airplane mode (on/off) until Firefox fixes it.

**Comment by followingwords** [comment]
Having same problem. On Pixel 4a, stock android v.10. This issue only came up upon updating Firefox yesterday.

When I look at an image's download location, it's says it was downloaded to download/download. In file app (both the stock one and a file manager app downloaded from the Play store) it does not appear in the image section, only when I go to the downloads folder. In neither file app does it show another folder named download within the download folder. If, for example on discord, I go to the option "file selector" instead of "image selector", the images appear there. 

Hopefully this info helps pin down the problem.

**Comment by Kimarnic** [comment]
> you can manually trigger a scan by toggling airplane mode (on/off) until Firefox fixes it.

That doesnt fix it, airplane mode doesnt work, still broken and I'll probably switch to Chrome because this is super annoying

**Comment by MusicOnline** [comment]
This bug is 7 months old by now, affecting a relatively important feature and a large amount of users as seen from the number of participants in this issue. The regression was also reported in this issue some time ago. 

Seeing the points above, hopefully the development team can responsibly prioritize this user-facing issue and not just have it in the backlog...

**Comment by Amejia481** [comment]
> This bug is 7 months old by now, affecting a relatively important feature and a large amount of users as seen from the number of participants in this issue. The regression was also reported in this issue some time ago.
> 
> Seeing the points above, hopefully the development team can responsibly prioritize this user-facing issue and not just have it in the backlog...

Sorry, for all the inconveniences and thanks for all the feedback this is in our backlog for this week. https://github.com/orgs/mozilla-mobile/projects/3

**Comment by black7375** [comment]
The same happens on the LG V40(Android 10).

**Comment by Amejia481** [comment]
It could be that it's a file specific issue see https://github.com/mozilla-mobile/fenix/issues/8754#issuecomment-690298123
Using the same download links a wasn't able to reproduce.

Could you please share the **links** that you have tried to download that are not getting into the "Downloads" folder + the file manager that you are using?

 We need:

- **Downloads links.** 
- **File manager app.**
- **Device and Android version.**

 Thanks in advance!

So far I have tested using:

Test links:

- https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf
- https://www.thinkbroadband.com/download

Devices:

- Samsung S8 with Android 9 with one UI version 1.0
- Motog5  with Android 7,  default android file browser
- Pixel 3 with Android 11, default android file browser
- Emulator Pixel 3a with Android 11
- Emulator Pixel 2 with Android 9
- Emulator Nexus one with Android 6

**Comment by Attoy** [comment]
> We need:
> 
>     * **Downloads links.**

https://unsplash.com/photos/1sim8ojvCbE
https://getwallpaper.com

>     * **File manager app.**

com.tohsoft.filemanager.v2 - version 1.40
 
>     * **Device and Android version.**

Samsung S20+ 5G, Android 10

> Thanks in advance!

Thank you!

**Comment by followingwords** [comment]
Downloads links.
* Images from: 
-Facebook
-weibo.com
-tumblr via app 
-tumblr via Firefox mobile

File manager app.
*default Android file app
*File Manager by ASUSTek
**please note, it is not just file manager that is not finding the images. Neither my text messaging app (Textra) nor discord can find them either. So it is *not* a file manager app issue. 

Device and Android version.
*Pixel 4a, android v 10, Aug 5th patch

**Comment by fersosam** [comment]
Hi, 

In my case the Samsung file manager does in fact find all the downloaded files. But then only if I dive down into "Internal Storage", they don't appear in the "Categories" shortcuts like "Documents" or "Downloads" But the Samsung Gallery doesn't ever show the files. It is not file specific, I can try any image or video on 9Gag for example. And I tried images at random from the web including one from Mozilla's own site:

https://cdn.vox-cdn.com/thumbor/1dQPFkvA1r1fiaV9gpSzfcqvAaM=/0x0:1920x1080/1520x0/filters:focal(0x0:1920x1080):format(webp):no_upscale()/cdn.vox-cdn.com/uploads/chorus_asset/file/8775207/thumbsupcontrols.jpg

https://www.economist.com/img/b/1280/720/90/sites/default/files/images/print-edition/20200718_WBD000_0.jpg

https://www.mozilla.org/media/contentcards/img/home-en/card_2/card_2-high-res.7f53d2324df6.jpg
 
Non of them show up in Samsung Gallery.

I'm on an S20+ with software update version G985FXXU4BTIB / G985FOWA4BTIB / G985FXXU4BTI6 Septembre 1, 2020. I tried just now with Firefox 81.1.1 (Build #2015764547) and Nightly 200929 17:11 (Build #2015766795).

**Comment by Amejia481** [comment]
> Downloads links.
> 
>     * Images from:
>       -Facebook
>       -weibo.com
>       -tumblr via app
>       -tumblr via Firefox mobile
> 
> 
> File manager app.
> *default Android file app
> *File Manager by ASUSTek
> **please note, it is not just file manager that is not finding the images. Neither my text messaging app (Textra) nor discord can find them either. So it is _not_ a file manager app issue.
> 
> Device and Android version.
> *Pixel 4a, android v 10, Aug 5th patch

@followingwords thanks for the clarification, it really helped me to understand the real issue :), I put a patch for it https://github.com/mozilla-mobile/android-components/pull/8586, it will be available in Fenix after it gets merged and Fenix updates to the new version of ac.

**Comment by fersosam** [comment]
Thank you @Amejia481 , I just tried it on Nightly 201003 17:09 (Build #2015767563) and it's fixed.

**Comment by black7375** [comment]
I also checked that it works.

**Comment by Djfe** [comment]
nice, is it possible to uplift this into the beta? (I don't know the dates of the release cycle)

**Comment by ZacharyGeorgeNN** [comment]
Will this issue be fixed in Daylight?

**Comment by Croman12** [comment]
Can confirm the issue is fixed, on Xiaomi MI 9T android 10 with latest Nightly. Great job!

**Comment by Amejia481** [comment]
> nice, is it possible to uplift this into the beta? (I don't know the dates of the release cycle)

We are going to added it to the patches be uplifted. Here you can see the Release management schedule.

**Comment by st3fan** [comment]
I would normally not pick up a patch like this so quickly for uplift. But the fact that @black7375 , @fersosam and @Croman12 confirmed this works for them now on Nightly gives me a some extra confidence. Thank you for testing!

**Comment by Djfe** [comment]
I just did a test, it's fixed for me as well now (in nightly)

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest Nightly build from 10/7.
Tested with the following devices:
* Google Pixel 4XL (Android 11),
* Samsung Galaxy S10+ (Android 10),
* Samsung Galaxy Tab S6 (Android 10), and
* Google Pixel (Android 10).

**Comment by cardshark414** [comment]
This is occurring on the latest nightly on my stock OP7Pro. OnePlus gallery will not display images downloaded through nightly. Google photos app on the same device seems to show them after several minutes.

**Comment by Djfe** [comment]
Which build are you on?
Settings -> About Firefox Nightly
![Screenshot_20201008-060655_Firefox_Nightly~2](https://user-images.githubusercontent.com/5685512/95413805-9a7c9900-092c-11eb-82dd-c4dfc50b196c.png)

**Comment by cardshark414** [comment]
2015768139

**Comment by Djfe** [comment]
That's unfortunate, so the fix didn't work for you?
Just to make sure: This fix only works for images, that you downloaded with the new version. You tested it, by downloading new images, right?

**Comment by cardshark414** [comment]
Yes. This is the first version of nightly im using. The issue is odd
though. Google messages doesnt see any of the images until i close it out
and reopen it. OnePlus gallery shows at least some of them after 5 to 10
minutes or if i close and reopen a few times. and google photos seems to
show at least a few of them after a couple minutes. and all of them after i
close and reopen it. i tested with .jpg and .png photos.

**Comment by byeuji** [comment]
Glad to see this is an active conversation -- I am having this issue on the Daylight build 2015768387, on a OnePlus 7T. 
Using a File Manager (in this case, ASUS File Manager 2.0), I was able to verify the images downloaded to the sdcard/downloads folder, but it was not visible in my share dialogs in other apps or in gallery. 
But if I used the File Manager to rename the file (from imagename.jpg to imagename.jpeg), the image immediately shows up.
I don't think the filetype is the thing that triggered the change -- I suspected the image file was being downloaded maybe lacking the MIME data the system needs to recognize it as a new image, and renaming the file somehow accomplished that.

The reason I suspected this was because this is a perennial problem problem I've had across many phones with Google Images, where the image downloads, but never shows up in system dialogs or folders. And the workaround I've used is to open the file through the Android download manager via Gallery, and then used the Share dialog to submit it back to myself over Pushbullet. The Pushbullet version then ends up in the downloads folder, and is visible to all apps.

I have always assumed that this situation was created because of how Google Images transmits the file data during download.

This is a lot of conjecture, but I hope this line of reasoning and other cases will help narrow down the issue.

**Comment by Novocastrian** [comment]
I'm also having the same issue on the latest Daylight (81.1.15 Build #2015769595) on my Galaxy Note 9. I tried downloading the first two images on this page (https://scryfall.com/search?q=thumb), but they don't appear in the Gallery app at all (under the Downloads folder).  When I open the Internal Storage > Download folder, I see each download attempt there. When downloading the same images from the same URL in Google Chrome, the images are downloaded to the same folder and are available to the Gallery app and others (Facebook Messenter, Discord, WhatsApp, etc.) immediately.

**Comment by Djfe** [comment]
Feedback Daylight 82.1.1: Works like a charm.
Thx, for fixing the bug.

@Novocastrian you should take a look whether playstore offers you an update to 82 already :)

**Comment by cardshark414** [comment]
Still having this issue in the latest nightly. Google photos seems to
display them, but only after close and reopen. It seems like Oneplus
gallery is not getting correct timestamp information for some of the
downloaded images. The ones it can see, get shuffled into the "download"
folder album in no particular order. Google messages seems like it can see
them all but only after close and reopen or waiting several minutes.

**Comment by Djfe** [comment]
'The ones it can see, get shuffled into the "download"'
do they have exif data? (different date for when they got shot)

**Comment by cardshark414** [comment]
They aren't camera captured images, theyre drawings. they do have "date
modified" property but Oneplus gallery does not display that. google images
shows them as possessing the modified date as the date they were
downloaded. The file manager does as well. Google Messages does not show
image dates from what i can see so i am unable to verify there.

**Comment by Novocastrian** [comment]
> 
> 
> Feedback Daylight 82.1.1: Works like a charm.
> Thx, for fixing the bug.
> 
> @Novocastrian you should take a look whether playstore offers you an update to 82 already :)

@Djfe, yep. Didn't realise the update was queued but not applied on my phone at the time of my last post. The latest version fixed this for me.

**Comment by damianlarson** [comment]
Hey all,

See the issue is closed, but still having similar problems to @cardshark414 on both Fenix Nightly and Daylight on my OnePlus 7T.

Images downloaded from Firefox are unable to be opened in the OnePlus File Manager--they open and then immediately close--and are not indexed until they are copied/cut and pasted into a different directory, where they're able to be opened and appear in the Gallery app as they should. Occurred with both JPG and PNG images. Tested against Chrome as well to see if it was a OnePlus issue, but Chrome images seem to index right away just fine.

Device: OnePlus 7T (running Oxygen OS v10.0.15)
Fenix Version: Daylight v83.1.0, Nightly v201126 17:04 (Build#2015777931)

**Comment by cardshark414** [comment]
Wish they would reopen this so that it would receive more attention. I have
no idea who to reach out to in order to make that happen though. and this
issue is preventing me from updating daylight for the last few months since
it is my main browser.

**Comment by Amejia481** [comment]
Thanks for reporting this issue @damianlarson and @cardshark414 I will ask the QA team to see if they are able to reproduce on your same devices.

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the 12/10 Nightly build with OnePlus 5T (Android 9), and OnePlus 6T (Android 9).
All the downloaded files were displayed in the three-dot menu-Downloads and could be opened without issues.

<img src="https://user-images.githubusercontent.com/40694158/101771815-2c527f00-3af3-11eb-863e-5abe73e36e8d.jpg" width="300" height="550" />

**Comment by Djfe** [comment]
@softvision-miralobontiu have you tested apps that rely on android's media storage functionality?
like the Android file manager, whatsapp, etc.

this bug is not about the 3 dot menu

this is about other apps that don't index the file system for media files themselves but rely on Android (the OS) to do so. Android itself needs apps to report about new files they download or create (Firefox) which doesn't seem to work properly here for some oneplus users (oxygen os)

example: send a file via mail,whatsapp,Facebook or whatever else you use.it shows you a way to select media but relies on media storage. images downloaded from Firefox might not show up for these users on oneplus devices

https://developer.android.com/guide/topics/providers/document-provider (some apps like my mail app rely on saf (storage access framework) for selecting files. images for example might not be listed in the image tab there if they weren't indexed correctly by Firefox

**Comment by softvision-miralobontiu** [comment]
Sorry for the confusion, I confirm that all the downloaded files can be seen on the OnePlus devices we've tested, in the media storage. All the files can be seen in the device's gallery, as seen below:

![Untitled](https://user-images.githubusercontent.com/40694158/101791333-14d3c000-3b0c-11eb-9740-f19fe39ae449.png)

**Comment by Djfe** [comment]
great :)

**Comment by cardshark414** [comment]
I just installed the 12/10 nightly build (Build #2015780619) and the issue
persists. i need to close-out and reopen the Google Messages app at least
once, and often multiple times for the downloaded images to appear when
attempting to send an image from the recent images list. Or simply waiting
5 or 10 minutes may cause they to finaoly show.

In the onePlus gallery app, the images don't even seem to make it into the
"recent" images list, but they do show up in the "download" album under
"app collections", just as if they were downloaded weeks or months ago much
farther down the list.

Even the Google Photos app does not immediately show the downloaded images.
I typically need to close and reopen it for them too appear. Though, they
do show as having been downloaded today in the Google Photos app, whereas
in the past, some may have shown to have been downloaded weeks or months
ago when they were in fact downloaded just moments before.

I do not experience this issue at all with the version of daylight I have
continued to use (68.11.0)

**Comment by Djfe** [comment]
> just as if they were downloaded weeks or months ago much
farther down the list.
Sounds to me like the images have a recorded date in their exif data and your gallery app sorts by the date an image was shot (not when the file was created)
I could be wrong though.
Do you know how to check the exif data of the particular image(s)?

**Comment by Djfe** [comment]
could you upload some of the Fotos here in a zip file, maybe? 
That would help a bit

**Comment by cardshark414** [comment]
Sure, I will upload images and screenshots shortly.

**Comment by damianlarson** [comment]
Hey @Djfe,

@cardshark414 may have additional screenshots, I thought I'd screen record so it's a little more clear exactly what's going on. Here's [a video](https://youtu.be/NlRmsaUSfMs) showing me trying to download a picture using Firefox Nightly 201213, obviously more recent than the build that was tested and validated. You'll see the same exact behavior is still occurring, files aren't appearing under "Recent", files aren't appearing in the Download section, and files aren't able to be opened; however, once I cut it and paste it somewhere else, suddenly it's indexed properly with the EXIF data and able to be opened and found in the Gallery.

Vice versa, I took [a video of me downloading the same exact image using Chrome](https://youtu.be/s_bIF6VeEP8), and there were no issues whatsoever. File had the EXIF data already, could be opened, and was found in Gallery without me having to do any work.

**Comment by Djfe** [comment]
I'm not one of the develeopers here, but the photo shows that the exif data says, that the photo was recorded in July 2020. Have you scrolled down to photos from July after downloading it in Nightly? Maybe it appears there.
Then it's up to your gallery app and the way it sorts images (by record date)

**Comment by Djfe** [comment]
does the problem occur for you with this image? 
https://upload.wikimedia.org/wikipedia/commons/thumb/2/28/JPG_Test.jpg/382px-JPG_Test.jpg

**Comment by damianlarson** [comment]
It does, yeah.

Even with this image, the file is unable to be opened in the File Manager, the file doesn't appear in Recents in the File Manager, and the file doesn't appear in Downloads in the Gallery until I cut and paste it to a different file location which triggers the indexing. And downloading the _exact same image_ in Chrome works just fine. It's not an issue with the Gallery app, as the File Manager also is unable to open the file.

Here's [another video using that JPG Test image which includes both Firefox Nightly and Chrome](https://youtu.be/MoWGHeUmMuk). Nightly fails, Chrome doesn't.

**Comment by Djfe** [comment]
it would be nice if someone could take a look at this again

even if it has been fixed for most devices

**Comment by Amejia481** [comment]
I can't compromise on a timeline but as soon as I have some time available I will take look.

**Comment by fennecbutt** [comment]
Oneplus 7 Pro Oxygen OS 10.0.10.GM21BA Firefox 84.1.0 (Build #2015780659)

I've had this issue all 2020. If I download an image in firefox, it will not appear in my gallery nor in apps that integrate the gallery. It will appear in certain file browsing apps like Google's files app.

I've figured out what causes it on my device. When a file is downloaded, I can find it in files and manually open it with the Gallery app. If I three dots->file details, I can see file name, resolution, size attributes but no time. This file will not appear in the Gallery app by itself. If I use the Files app to "move" the file to the same folder (downloads) then it appears in the Gallery app (and all other apps it was missing from) and if I three dots->file details, it now has a "time" attribute along with the others.

I have a feeling this is because either: certain apps expect the time attribute (file creation date?) and do not display an image if it is not present, _or_ certain apps will order files missing this attribute last (although Files by Google lists it first) when ordering by date.

I can cause the missing files to appear if I move the file, or restart my phone. But the files always gain the file time attribute in order for them to appear.

**Comment by acneaguu** [comment]
> 
> 
> Oneplus 7 Pro Oxygen OS 10.0.10.GM21BA Firefox 84.1.0 (Build #2015780659)
> 
> I've had this issue all 2020. If I download an image in firefox, it will not appear in my gallery nor in apps that integrate the gallery. It will appear in certain file browsing apps like Google's files app.
> 
> I've figured out what causes it on my device. When a file is downloaded, I can find it in files and manually open it with the Gallery app. If I three dots->file details, I can see file name, resolution, size attributes but no time. This file will not appear in the Gallery app by itself. If I use the Files app to "move" the file to the same folder (downloads) then it appears in the Gallery app (and all other apps it was missing from) and if I three dots->file details, it now has a "time" attribute along with the others.
> 
> I have a feeling this is because either: certain apps expect the time attribute (file creation date?) and do not display an image if it is not present, _or_ certain apps will order files missing this attribute last (although Files by Google lists it first) when ordering by date.
> 
> I can cause the missing files to appear if I move the file, or restart my phone. But the files always gain the file time attribute in order for them to appear.

I think this is indeed the cause. I've been able to reproduce this on my OP 7T Pro with Oxygen OS 10.0.13.HD01BA using Firefox 84.1.4 (Build #2015785883). If I download the same image using Google Chrome (87.0.4280.141), the image gets a time and date and is now seen by the stock gallery app.

**Comment by MrBrax** [comment]
do different distros of android handle filesystems differently or something? this seems like a very strange thing to alter default behaviors.. for me, firefox mobile has worked since maybe june or so

**Comment by monkeber** [comment]
Able to reproduce this issue on OnePlus 7T Pro Oxygen OS 10.0.13.HD01AA, Firefox version 84.1.4, images downloaded through firefox won't show up in Gallery, Google Photos and Telegram, if I donwload the same image through Google Chrome in the same folder (/storage/emulated/0/Download) - everything shows up fine.

Had this issue last month on Xiaomi Mi A1, but can't reproduce it there anymore.

**Comment by Mysterius** [comment]
Still not fixed (OnePlus 7T pro). Haven't been able to save a single picture in the whole fucking year. I won't wait for 2021 to pass. Switching back to Chrome, a browser's first feature is for it to be working as expected.

**Comment by cadeyrn** [comment]
I have a OnePlus 7T Pro as well and can **not** reproduce the issue. I downloaded an image and it's shown by the Google Photos app and by the OnePlus Gallery app. Please make sure that you're using the latest version of your OS and of the OnePlus apps.

**Comment by byeuji** [comment]
Following up: I continue to have this issue occur. It happened just a couple days ago, though I can't recall exactly what occurred. Essentially I downloaded an image off a website in Firefox and had to push it through Pushbullet to myself to get it to be shareable via app share actions.

Firefox 84.1.4 (Build #2015785883)
OnePlus 7T model HD1907
Manufacturer stock OS, version 10.0.12.HD63CB (system reports up to date)

**Comment by cardshark414** [comment]
This issue is still occurring on my device as well (7 Pro OOS 10.3.5
GM21AA). Considering this issue does not occur with any other browsers or
with the older version of Firefox (68.11.0), I don''t think the fix will be
to just update the OS.

**Comment by Nicocchi** [comment]
I have had this issue since I have gotten my phone, One Plus 7 Pro (Oxygen OS 10.3.7) in September 2020. This issue is ongoing till today with any image I download off of Firefox (84.1.4). If I download images from any other browser (Chrome or Brave) they appear in default gallery, Discord and messenger. But Firefox does downloads do not. In Discord's case, if I open attachments then I can see it, but the default gallery picker I cannot. The default gallery app doesn't show any of the downloads from Firefox. However I do use another Gallery app called Gallery that does show Firefox downloads. I haven't tested nightly but I have tried Firefox Beta and this happens on it as well. I have also reset my OS and still have the same problems.

**Comment by Mugurell** [comment]
Seems like we have to `addCompletedDownload` that we use was deprecated on Android 29 and we have to use `MediaStore.Downloads` going forward
https://developer.android.com/reference/android/app/DownloadManager#addCompletedDownload(java.lang.String,%20java.lang.String,%20boolean,%20java.lang.String,%20java.lang.String,%20long,%20boolean,%20android.net.Uri,%20android.net.Uri)

**Comment by Amejia481** [comment]
> `addCompletedDownload`

It should be running only on devices that [opt in to use scoped storage](https://github.com/mozilla-mobile/android-components/blob/a4bcd420a01898e5207269f0bd9119987fa4f4b2/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L424), as Fenix is opting in out (For now)  we are executing it, when this [PR](https://github.com/mozilla-mobile/fenix/pull/17701) lands the situation should improve on nightly for devices with Android 10 and above.

**Comment by Mugurell** [comment]
This is now fixed in the current Nightly. But only for fresh installs.
There is still a small issue to be resolved with upgraded users. In progress.

**Comment by Croman12** [comment]
Since a few weeks (months) this was fixed for a lot of users, including me. Now since version 2021.02.10 (might be a version earlier) this doesn't work anymore. I'm on Android 10 (HAVOC OS), Xiaomi MI 9T. Anyone else experiencing this problem again since a few days?

**Comment by Djfe** [comment]
the first fix still works fine on my device (no regression on the oneplus 3t Havoc os Android 10)
I tested the current release, beta and nightly clients

**Comment by Mugurell** [comment]
> A few weeks (months) ago this was fixed for a lot of users, including me. Now since version 2021.02.10 (might be a version earlier) this doesn't work anymore. I'm on Android 10 (HAVOC OS), Xiaomi MI 9T. Anyone else experiencing this problem again since a few days?

Can you please offer more details about the issue you're encountering as this should be fixed in the latest Nightly?
- version of the application
- device name and Android version
- whether this is a general problem or it happens only for specific file types downloaded / downloads from specific websites

Anything that could help us reproduce the issue.

**Comment by Amejia481** [comment]
In addition to the information above a print screen of the about Firefox screen will be really useful.
<img width="947" alt="image" src="https://user-images.githubusercontent.com/773158/107780188-5ce1af80-6d14-11eb-8ec0-553ab3e77a73.png">

**Comment by acneaguu** [comment]
> 
> 
> This is now fixed in the current Nightly. But only for fresh installs.
> There is still a small issue to be resolved with upgraded users. In progress.

I've tried Nightly (Build #2015792907) for the last week or so and this has fixed the issue for me: when downloading an image, it shows up in my _Downloads_ gallery folder as expected. Downloads from the regular Firefox browser (85.1.3 Build #2015792283) do not show up in the _Downloads_ folder in the device gallery. My device is the OnePlus 7T Pro running Android 10 with Oxygen OS 10.0.13.HD01BA. 

Thanks for fixing it in Nightly and I'm looking forward to this becoming part of the regular Firefox as well. For now, I'll stick with Nightly.

**Comment by Croman12** [comment]
> 
> 
> > A few weeks (months) ago this was fixed for a lot of users, including me. Now since version 2021.02.10 (might be a version earlier) this doesn't work anymore. I'm on Android 10 (HAVOC OS), Xiaomi MI 9T. Anyone else experiencing this problem again since a few days?
> 
> Can you please offer more details about the issue you're encountering as this should be fixed in the latest Nightly?
> 
>     * version of the application
> 
>     * device name and Android version
> 
>     * where do you check if the file is indexed or not? In a particular gallery app / file explorer app?
> 
>     * whether this is a general problem or it happens only for specific file types downloaded / downloads from specific websites
> 
> 
> Anything that could help us reproduce the issue.

It looks like it's a problem on my end. I tried with recent versions of Chrome, Edge and the default Android browser preinstalled on my phone. Only images en videos are being indexed under 'Recent' in my 'Files' app (com.android.documentsui). No idea how to fix this, really. While it not seems to be related to Fenix, anyone got any idea on what the problem is or how to solve it? Only other solution I can think of at this point is a complete reinstall of my OS. Help would be much appreciated!

**Comment by Croman12** [comment]
So I finally had some time to take a closer look at the issue. I knew that version 2021.02.04 worked for me before, but when I installed the version using a clean install, the issue was still there. This led me to think my phone was causing the issue, but the default browser did work like a charm. I went al the way back to when I first reported the issue was fixed (version 2020.10.06) and it indeed worked like it should. I manually upgraded all the way to version 2021.02.01 and here is where things get interesting.

A clean install of version 2020.02.01 (February 1st) works like it should. Recently downloaded files of pretty much any filetype show up in the recent files section of the default Files-app (com.android.documentsui). If I perform a clean install of version 2021.02.02 it does not work anymore. A fresh install of versions 2021.02.02 up to 2021.02.06 have the issue.

But if I upgrade from previous versions (one version at a time), it works.  up to 2021.02.04. Meaning that when I upgrade from 2021.02.01 to 2021.02.02 is does work, all the way up to 2021.02.04, the issue is gone and it actually works like it should. but when I upgrade to 2021.02.05 the issue is back.

Some interesting notes:
- What I mean with 'issue' is that files don't seem to be indexed and shown under recent files. This excludes filetypes from images and videos. So every version I tested only showed pictures and videos under recent files.
- When the documents (pdf, docx, apk, nzb etc...) don't get indexed, one thing stood out to me. When I downloaded pdf files, and want to open them directly within Firefox using the notification that appears, two different PDF-readers fail to open the files, stating there is some reading error and they don't have access to the file.
- I can see that around that time the issue came back, some changes have been made to actually solve the issue for other users. Maybe something changed tot the path where downloads are stored or the way downloads are handled is preventing other apps from accessing the files, failing to index them or open them (they simply can't see the files)?

I'm running Android 10 (Havoc 3.8) on my Xiaomi Mi 9T. Any further questions, please let me know.

**Comment by Djfe** [comment]
you could use mozregression to bisect automatically :) (faster and easier than doing it all manually)
https://mozilla.github.io/mozregression/quickstart.html

It even allows bisecting down to commit level.
(Though it deletes app data on every reinstall of the app afaik. so you can't simulate upgrades in automated fashion)

**Comment by kvahid** [comment]
I observe the same problem with current version

**Comment by Mugurell** [comment]
> I observe the same problem with current version

To be able to reproduce this issue we'll need more info:
Please provide:
- exact version of the app used
- whether it's a new install or an upgrade 
(would be awesome if you could try the current Nightly with a fresh install)
- device model and Android version. Official or custom rom.
- the type of file downloaded
- in which application you expect to see the downloaded file but can't

**Comment by SSchott** [comment]
I don't know if anyone has mentioned this already, but a similar things has happened for years in the desktop version (where is way more difficult to identify, given the way people use the devices). Is there a deeper underlying cause to this behavior?

**Comment by Djfe** [comment]
since there is no desktop version of Android: which os's media index are you referring to?

**Comment by SSchott** [comment]
> since there is no desktop version of Android: which os's media index are you referring to?

I use Mendeley as a Reference Manager, and noticed that whenever I download a PDF with Firefox, the "automatic scanning" of Mendeley doesn't index the file. This does not happen with any other web browser. Thought it was curious that pretty much the same happens in the desktop version... (Windows 10 and Linux btw).

**Comment by Croman12** [comment]
I installed the version from yesterday (Nightly April 19th), the problem is still there. I did a clean install. For the moment I'm staying on version February 1st. Is this still being looked at?

**Comment by Mugurell** [comment]
@Croman12 Do you have examples of files downloaded (links of) but then not available in file managers?
Another very important detail is the application in which these files are not available and whether or not the files appear after a device restart.
This issue seems to reproduce for just a few users/devices so it would help us greatly having more details about the exact scenario for it.

**Comment by Croman12** [comment]
> @Croman12 Do you have examples of files downloaded (links of) but then not available in file managers?
> Another very important detail is the application in which these files are not available and whether or not the files appear after a device restart.
> This issue seems to reproduce for just a few users/devices so it would help us greatly having more details about the exact scenario for it.
> // A logcat of the moment when download finishes would also help. For details on how to get this - https://github.com/mozilla-mobile/fenix/wiki/Logging-Crash-Information (in this case we'd need the log for the moment when the download finishes. A few seconds before, a few seconds after (instead of the crash)).

Situation is still the same as my comment on February 18th of this issue. Firefox does not crash, I'm not sure if you still want a logcat then? The files are all document files (at least any pdf, nzb and torrent files that I've tried). The feature broke within a couple of versions as I said in my post. So changes during that time span are probably where we need to start looking. Do you still want the logcat?

**Comment by Mugurell** [comment]
@Amejia481 I saw you recently improved the flow of us opening files after download on A10.
Do you know if anything can be done to open 3rd party apps too?

**Comment by Amejia481** [comment]
Yeah, that one was just to fix an issue on Android 10 when opening files, unfortunately  I'm not sure what we can do to improve the actual situation for 3rd party apps, we have to investigate further.

**Comment by Mugurell** [comment]
I tested downloading with a Samsung S9 with Android 10 and then checking files in the "Download" folder using Solid Explorer:

1. https://www.thinkbroadband.com/download
- Chrome stable
   - tapping any download link does not actually starts the download
   - copying a link and opening it in a new tab will download the file which is properly indexed and can't be opened (probably garbled data)
- Fenix Nightly
   - tapping any download link will start the download
   - the file is properly indexed but can't be opened (probably garbled data)
   
2 . https://www.futurehosting.com/network-test-files/
- Chrome stable
   - tapping any download link does not actually starts the download
   - copying a link and opening it in a new tab will download the file which is properly indexed and can be opened 
- Fenix Nightly
   - tapping any download link will start the download
   - the file cannot be opened from the "Download completed" dialog but is properly indexed and can be opened from Solid Explorer.
   
3 . http://www.ovh.net/files
- Chrome stable
   - tapping any download link will actually start the download
   - the file is available in Solid Explorer but can't be opened (probably garbled data).
- Fenix Nightly
   - tapping any download link will actually start the download
   - the file is available in Solid Explorer but can't be opened (probably garbled data).
   
For me on this device the downloads were always properly indexed so it's good to know that this is not affecting all Android 10 devices but I don't know how many devices are affected.
Asking if QA can also help with other devices with Android 10, other download locations and a comparison with other browsers.

**Comment by lobontiumira** [comment]
I have tested on OnePlus 5T (Android 10), and Google Pixel (Android 10) devices, on the 4/28 Nightly build, test page: file-examples.com, and some images from different pages.

**OnePlus 5T (Android 10)** displayed all the downloaded files on the "Download" page in Fenix, and on the "Download" folder of the device. 
* Fenix Nightly:
-- tapping on the download link, asks the user to allow the download, then starts the download,
-- tapping on the "Open" button from the "Download complete" prompt, opens the file.
![OnePlus](https://user-images.githubusercontent.com/40694158/116382651-d02c9500-a81e-11eb-8fd4-93fbea80c7d9.png)
* Chrome:
-- tapping on the video download link, asks the user to allow the download, then starts the download,
-- tapping on the pdf download link, opens the pdf directly.

**Google Pixel (Android 10)** displayed all the downloaded files on the "Download" page in Fenix, and on the "Download" folder of the device.  
* Fenix Nightly:
-- tapping on the download link, asks the user to allow the download, then starts the download,
-- tapping on the "Open" button from the "Download complete" prompt, opens the file.
* Chrome:
-- tapping on the video download link, asks the user to allow the download, then starts the download,
-- tapping on the mp4 download link, opens the file directly,
-- tapping on the pdf download link, opens the pdf directly.

On both devices, all the downloaded files, from Fenix and Chrome, were displayed on the device's "Download" folder.
I'll remove the qa:needed label.

**Comment by tegutistgut** [comment]
Is my issue related to this thread?

My issue: when downloading a .m3u file, Firefox (Android of course) does not download it in my download folder, but rather in the Music folder (_/storage/emulated/0/Music_)

Any other files are downloaded in the _/storage/emulated/0/Download_ folder

**Comment by Mugurell** [comment]
> Is my issue related to this thread?
> 
> My issue: when downloading a .m3u file, Firefox (Android of course) does not download it in my download folder, but rather in the Music folder (_/storage/emulated/0/Music_)
> 
> Any other files are downloaded in the _/storage/emulated/0/Download_ folder
> 
> Idk, maybe it happend with other file types too. This is annoying, because i cant change it and i cant find my files

Hello @tegutistgut and sorry for the late reply.
Your issue seems separate from the one discussed here.
If it still reproduces please open a new ticket while providing more details about the device you're using, it's Android version and an url example from which a downloaded m3u does not appear in the expected folder. A video showing the entire scenario in which this reproduces would be great since it would allow us to follow the exact same steps to reproduce and then work on fixing the issue.

**Comment by Mugurell** [comment]
Regarding the original issue, that downloaded files are not properly indexed based on our tests it seems to be a device/app (file manager / galerry) specific issue that we have little control over.

Removing my assignement but the ticket can still be commented on, maybe we can isolate the bug.

An actionable scenario would be if a certain download from other browsers appear as expected but not when being downloaded in Firefox.

**Comment by djdisodo** [comment]
original issue seems solved; i don't want to see any other issues commented here so closing
